diff --git a/lib/utils/checkAgainstRule.cjs b/lib/utils/checkAgainstRule.cjs
index 862f0c77f2ec60a26ac670230c531be88cead26f..e2c40a2009941ccd9e6c4a8f49f654c8e6ae1ef3 100644
--- a/lib/utils/checkAgainstRule.cjs
+++ b/lib/utils/checkAgainstRule.cjs
@@ -45,7 +45,7 @@ async function checkAgainstRule(options, callback) {
 	const [primary, secondary] = settings;
 	const ruleFunc = rule(primary, secondary || {}, context);
 
-	ruleFunc(
+	await ruleFunc(
 		root,
 
 		// NOTE: This temporary PostCSS result doesn't have a property for Stylelint use.
diff --git a/lib/utils/checkAgainstRule.mjs b/lib/utils/checkAgainstRule.mjs
index c82c6c82734ba7a41f59a57aa7a89f149378599f..168270665e953635e3cb223fe5ebdbbbaf69ff54 100644
--- a/lib/utils/checkAgainstRule.mjs
+++ b/lib/utils/checkAgainstRule.mjs
@@ -42,7 +42,7 @@ export default async function checkAgainstRule(options, callback) {
 	const [primary, secondary] = settings;
 	const ruleFunc = rule(primary, secondary || {}, context);
 
-	ruleFunc(
+	await ruleFunc(
 		root,
 
 		// NOTE: This temporary PostCSS result doesn't have a property for Stylelint use.
